webapp/components/_new/generic/BaseInput/BaseInput.vue

```diff
@@ -0,0 +1,194 @@
+<template>
+   <div class="input-wrapper" :class="[focus]">
+      <base-icon class="input-icon" v-if="icon" :name="icon" /> 
+      
+      <input
+         class="base-input"
+         :class="[
+            icon && `input-has-icon`,
+            iconRight && `input-has-icon-right`,
+            iconRightSecondary && `input-has-icon-right-secondary`
+         ]"
+         :id="id"
+         :name="name ? name : model"
+         :type="[showPassword ? 'text' : 'password']"
+         :v-model="type"
+         :autofocus="autofocus"
+         :placeholder="placeholder"
+         :tabindex="tabindex"
+         :disabled="disabled"
+         :readonly="readonly"
+         :value.prop="innerValue"
+         @input="handleInput"
+         @focus="handleFocus"
+         @blur="handleBlur"
+         v-on:keydown.caps-lock="caps = true"
```

Comment:
This is too specific for the password type input to have it in the input component.